transport/internal/dca/dca.go

```diff
@@ -129,11 +129,17 @@ func getMTLSMode() string {
 	return strings.ToLower(mode)
 }
 
-func mergeEndpoints(base, newHost string) (string, error) {
-	u, err := url.Parse(base)
+func mergeEndpoints(baseURL, newHost string) (string, error) {
+	u, err := url.Parse(fixScheme(baseURL))
 	if err != nil {
 		return "", err
 	}
-	u.Host = newHost
-	return u.String(), nil
+	return strings.Replace(baseURL, u.Host, newHost, 1), nil
```

Comment:
This is a bit different than what is returned before. Is this intentional? See docs for [URL.String()](https://golang.org/pkg/net/url/#URL.String) for more details.

Reply:
Yes, this part is intentional. The problem with using u.String() is that it will inject a "//" to the beginning of the string even if I explicitly clear the scheme (i.e. we end up with "//myhost:port/foo/bar" when the baseURL does not have a scheme) The revised logic makes our intention very explicit, which is to replace default host with the new host, and avoids idiosyncrasies with URL.String(). Let me know if you have a better solution. Thanks!